packages/x-data-grid/src/models/gridBaseSlots.ts

```diff
@@ -22,6 +22,18 @@ export type MenuItemProps = {
   value?: number | string | readonly string[];
 };
 
+export type ToolbarProps = {
+  ref?: React.Ref<HTMLDivElement>;
+  role?: string;
+  'aria-orientation'?: string;
+};
+
+export type ToolbarButtonProps = {
+  ref?: React.Ref<HTMLButtonElement>;
+  tabIndex?: number;
+  onKeyDown?: React.KeyboardEventHandler<HTMLButtonElement>;
+};
+
```

Comment:
Does this really belong here? This is for base slot props, I would place slot props in a different location. See base slot vs slot difference here: https://github.com/mui/mui-x/blob/d85fff19138b109a8d81655d55de62dbca2c6065/packages/x-data-grid/src/models/gridSlotsComponentsProps.ts#L92-L156

Reply:
If you want a test to classify a slot as base or not, "base" slots are what every non-material design-system will need to re-implement to use the grid, so things like buttons, selects, etc.

Reply:
Removed this, was experimenting with having the styled versions of the components as base slots but it felt like a dirty workaround.